src/ManipulationMsg.chpl

```diff
@@ -1063,36 +1063,36 @@ module ManipulationMsg {
     const name = msgArgs.getValueOf("name"),
           axis = msgArgs.get("axis").getPositiveIntValue(nd),
           numReturnArrays = msgArgs.get("numReturnArrays").getIntValue(),
-          rnames = [i in 0..<numReturnArrays] st.nextName();
+          rnames = for 0..<numReturnArrays do st.nextName();
 
     var gEnt: borrowed GenSymEntry = getGenericTypedArrayEntry(name, st);
 
     proc doUnstack(type t): MsgTuple throws {
       const eIn = toSymEntry(gEnt, t, nd),
-            (valid, shapeOut) = unstackedShape(eIn.tupShape, axis, numReturnArrays);
+            shapeOut = unstackedShape(eIn.tupShape, axis);
 
-      if !valid {
-        const errMsg = "Unable to unstack array with shape %? along axis %? into %? arrays".doFormat(eIn.tupShape, axis, numReturnArrays);
+      if eIn.tupShape[axis] != numReturnArrays {
+        const errMsg = "Cannot unstack array with shape %? along axis %? into %? arrays".doFormat(eIn.tupShape, axis, numReturnArrays);
         mLogger.error(getModuleName(),pn,getLineNumber(),errMsg);
         return new MsgTuple(errMsg,MsgType.ERROR);
-      } else {
-        var eOuts = for rn in rnames do (try st.addEntry(rn, (...shapeOut), t));
+      }
 
-        // copy the data from the input array to the output arrays while unstacking
-        for arrIdx in 0..<numReturnArrays {
-          forall idx in eOuts[arrIdx].a.domain with (
-            var agg = newSrcAggregator(t),
-            const imap = new unstackIdxMapper(nd, arrIdx, axis)
-          ) {
-            const inIdx = imap(if nd == 2 then (idx,) else idx);
-            agg.copy(eOuts[arrIdx].a[idx], eIn.a[inIdx]);
-          }
-        }
+      var eOuts = for rn in rnames do (try st.addEntry(rn, (...shapeOut), t));
 
-        const repMsg = try! '+'.join([rn in rnames] "created " + st.attrib(rn));
-        mLogger.info(getModuleName(),pn,getLineNumber(),repMsg);
-        return new MsgTuple(repMsg, MsgType.NORMAL);
+      // copy the data from the input array to the output arrays while unstacking
+      for arrIdx in 0..<numReturnArrays {
+        forall idx in eOuts[arrIdx].a.domain with (
+          var agg = newSrcAggregator(t),
+          const imap = new unstackIdxMapper(nd, arrIdx, axis)
+        ) {
+          const inIdx = imap(if nd == 2 then (idx,) else idx);
```

Comment:
is this hardcoded 2 what we want or do we want like greater than 1? (Don't understand enough to know, just an actual question)

Reply:
Yeah it's supposed to be a `2` because when `nd==1`, the output array's domain will have a rank of `nd-1`. So this is the same thing as saying `if eOuts[arrIdx].a.domain.rank == 1 then (idx, ) else idx`. I couldn't decide which was less confusing, so lmk if you think it should be changed.